File: app/src/main/java/com/andraganoid/playsomemedia/model/VideoDao.java
package com.andraganoid.playsomemedia.model;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface VideoDao {

    @Query("SELECT * FROM video_table ORDER BY dateTaken ASC")
    LiveData <List <Video>> getAllVideos();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Video video);

    @Query("DELETE FROM video_table")
    void deleteAllVIdeo();


}

File: app/src/main/java/com/andraganoid/playsomemedia/k/db/VideoDao.kt
package com.andraganoid.playsomemedia.k.db

import androidx.lifecycle.LiveData
import androidx.room.*
import com.andraganoid.playsomemedia.model.Video

@Dao
interface VideoDao {
    @Query("SELECT * FROM video_table ORDER BY dateTaken ASC")
    fun getAllVideos(): LiveData<List<Video?>?>?

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(video: Video?)

    @Query("DELETE FROM video_table")
    fun deleteAllVideos()

    @Delete
    fun deleteVideo(video: Video?)
}